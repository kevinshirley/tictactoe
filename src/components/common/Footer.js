import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import Fade from 'react-reveal/Fade';
import ks from '../../assets/kevinshirley.png';
import tictactoe from '../../assets/logo.svg';

class Footer extends Component {

  constructor() {
    super();

    var today = new Date(), year = today.getFullYear();

    this.state = {
      year: year
    };
  }

  render() {
    return (
      <footer className="footer">

        <div className="inner-footer">

          <div className="contact-social">

            <div className="social">
              <ul>
                <Fade bottom><li><a href="https://github.com/kevinshirley/tictactoe" target="_blank" rel="noopener noreferrer"><i className="fab fa-github"></i></a></li></Fade>
                <Fade bottom><li><a href="mailto:info@kevinshirley.com"><i className="far fa-envelope"></i></a></li></Fade>
              </ul>
            </div>

          </div>

          <div className="copyrightandall">

            <div className="copyright">
              <Fade bottom><small>&copy; {this.state.year} <a href="http://kevinshirley.com" target="_blank" rel="noopener noreferrer">KevinShirley.com</a></small></Fade>
            </div>

            <div className="logo">
              <Fade bottom><Link to="/"><img src={tictactoe} alt="Tic Tac Toe Logo" /></Link></Fade>
            </div>

            <div className="poweredby">
              <Fade bottom><small>Powered By<a href="http://kevinshirley.com" target="_blank" rel="noopener noreferrer"><img src={ks} alt="Kevin Shirley" /></a></small></Fade>
            </div>

          </div>

          <div className="copyrightandall-mobile">

            <div className="poweredby">
              <Fade bottom><small>Powered By<a href="http://kevinshirley.com" target="_blank" rel="noopener noreferrer"><img src={ks} alt="Kevin Shirley" /></a></small></Fade>
            </div>

            <div className="copyright">
              <Fade bottom><small>&copy; {this.state.year} <a href="http://kevinshirley.com">KevinShirley.com</a></small></Fade>
            </div>

            <div className="logo">
              <Fade bottom><Link to="/"><img src={tictactoe} alt="tictactoe Logo" /></Link></Fade>
            </div>

          </div>

        </div>

      </footer>
    );
  }
}

export default Footer;