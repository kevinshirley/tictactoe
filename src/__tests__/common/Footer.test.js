import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../../components/common/Footer';
import ks from '../../assets/kevinshirley.png';

describe('Footer', () => {
  it('should render correctly in "debug" mode', () => {
    const footer = shallow(<Footer debug />);
  
    expect(footer).toMatchSnapshot();
  });

  // it('renders two ks images', () => {
  //   const footer = shallow(<Footer debug />);
  //   expect(footer.find(ks)).toExist();
  // });
});