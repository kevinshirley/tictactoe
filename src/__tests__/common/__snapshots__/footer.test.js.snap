// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render correctly in "debug" mode 1`] = `
<footer
  className="footer"
>
  <div
    className="inner-footer"
  >
    <div
      className="contact-social"
    >
      <div
        className="social"
      >
        <ul>
          <Fade
            bottom={true}
          >
            <li>
              <a
                href="https://github.com/kevinshirley/tictactoe"
                rel="noopener noreferrer"
                target="_blank"
              >
                <i
                  className="fab fa-github"
                />
              </a>
            </li>
          </Fade>
          <Fade
            bottom={true}
          >
            <li>
              <a
                href="mailto:info@kevinshirley.com"
              >
                <i
                  className="far fa-envelope"
                />
              </a>
            </li>
          </Fade>
        </ul>
      </div>
    </div>
    <div
      className="copyrightandall"
    >
      <div
        className="copyright"
      >
        <Fade
          bottom={true}
        >
          <small>
            © 
            2019
             
            <a
              href="http://kevinshirley.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              KevinShirley.com
            </a>
          </small>
        </Fade>
      </div>
      <div
        className="logo"
      >
        <Fade
          bottom={true}
        >
          <Link
            replace={false}
            to="/"
          >
            <img
              alt="Tic Tac Toe Logo"
              src=""
            />
          </Link>
        </Fade>
      </div>
      <div
        className="poweredby"
      >
        <Fade
          bottom={true}
        >
          <small>
            Powered By
            <a
              href="http://kevinshirley.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Kevin Shirley"
                src=""
              />
            </a>
          </small>
        </Fade>
      </div>
    </div>
    <div
      className="copyrightandall-mobile"
    >
      <div
        className="poweredby"
      >
        <Fade
          bottom={true}
        >
          <small>
            Powered By
            <a
              href="http://kevinshirley.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="Kevin Shirley"
                src=""
              />
            </a>
          </small>
        </Fade>
      </div>
      <div
        className="copyright"
      >
        <Fade
          bottom={true}
        >
          <small>
            © 
            2019
             
            <a
              href="http://kevinshirley.com"
            >
              KevinShirley.com
            </a>
          </small>
        </Fade>
      </div>
      <div
        className="logo"
      >
        <Fade
          bottom={true}
        >
          <Link
            replace={false}
            to="/"
          >
            <img
              alt="tictactoe Logo"
              src=""
            />
          </Link>
        </Fade>
      </div>
    </div>
  </div>
</footer>
`;
